sycl/unittests/scheduler/CommandsWaitForEvents.cpp

```diff
@@ -0,0 +1,107 @@
+#include "SchedulerTest.hpp"
+#include "SchedulerTestUtils.hpp"
+#include <helpers/PiMock.hpp>
+
+using namespace cl::sycl;
+
+struct TestCtx {
+  queue &Q1;
+  queue &Q2;
+
+  std::shared_ptr<detail::context_impl> Ctx1;
+  std::shared_ptr<detail::context_impl> Ctx2;
+
+  pi_event EventCtx1 = reinterpret_cast<pi_event>(0x01);
+  pi_event EventCtx2 = reinterpret_cast<pi_event>(0x02);
+
+  bool EventCtx1WasWaited = false;
+  bool EventCtx2WasWaited = false;
+
+  TestCtx(queue &Queue1, queue &Queue2)
+      : Q1(Queue1), Q2(Queue2), Ctx1{detail::getSyclObjImpl(Q1.get_context())},
+        Ctx2{detail::getSyclObjImpl(Q2.get_context())} {}
+};
+
+std::unique_ptr<TestCtx> TestContext;
+
+pi_result waitFunc(pi_uint32 N, const pi_event *List) {
+  EXPECT_EQ(N, 1u) << "piEventsWait valled for different contexts\n";
```

Comment:
```suggestion
  EXPECT_EQ(N, 1u) << "piEventsWait called for different contexts\n";
```
Minor, can be addressed separately